graphjet-core/src/main/java/com/twitter/graphjet/algorithms/TweetAuthorFilter.java

```diff
@@ -65,6 +68,6 @@ public void resetFilter(RecommendationRequest request) {}
    */
   @Override
   public boolean filterResult(long resultNode, SmallArrayBasedLongToDoubleMap[] socialProofs) {
-    return !authoredTweets.contains(resultNode);
+    return hasAuthors && !authoredTweets.contains(resultNode);
```

Comment:
why add the extra hasAuthors variable?
it looks like we just need to check !authoredTweets.isEmpty() && !authoredTweets.contains(resultNode);